project1-sps/script.js

```diff
@@ -1,4 +1,177 @@
+    (userSPS == "reversed paper" && pcSPS == "reversed stone") ||
+    (userSPS == "reversed stone" && pcSPS == "reversed scissors")
+  ) {
+    var winOrLose = "You lose! ";
+    userWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  // you lose scenario
+  else if (
+    (userSPS == "reversed paper" && pcSPS == "reversed scissors") ||
+    (userSPS == "reversed scissors" && pcSPS == "reversed stone") ||
+    (userSPS == "reversed stone" && pcSPS == "reversed paper")
+  ) {
+    var winOrLose = "You win! ";
+    pcWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  return winOrLose;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var userSPS = input;
+  // change username only for first time input
+  message = usernameInputCheck(userSPS);
+
+  // check for error input (for 2nd inputs onwards)
+  if (usernameInputCount > 1) {
+    //Defining game logic
+    var winOrLose = "Nothing happened. ";
+
+    // pc chooses
+    if (
+      userSPS == "reversed scissors" ||
+      userSPS == "reversed paper" ||
+      userSPS == "reversed stone"
+    ) {
+      var pcSPS = chooseSPS();
+      if (pcSPS == 1) {
+        var pcSPS = "reversed scissors";
+      } else if (pcSPS == 2) {
+        var pcSPS = "reversed paper";
+      } else if (pcSPS == 3) {
+        var pcSPS = "reversed stone";
+      }
+      winOrLose = winOrLoseReversedSPSLogic(userSPS, pcSPS);
+    } else if (
+      userSPS == "scissors" ||
+      userSPS == "paper" ||
+      userSPS == "stone"
+    ) {
+      var pcSPS = chooseSPS();
+      if (pcSPS == 1) {
+        var pcSPS = "scissors";
+      } else if (pcSPS == 2) {
+        var pcSPS = "paper";
+      } else if (pcSPS == 3) {
+        var pcSPS = "stone";
+      }
+      winOrLose = winOrLoseSPSLogic(userSPS, pcSPS);
+    }
+    // compare who wins
+
+    // check every scenario case
+    // if (input == "scissors"){
+    //   if (pcSPS == "scissors") {
+    //    var winOrLose = "It's a draw. ";
+    //   } else if (pcSPS == "paper") {
+    //    var winOrLose = "You win! ";
+    //  } else if (pcSPS == "stone") {
+    //    var winOrLose = "You lose. ";
+    //  }
+    //}
+
+    roundsPlayed += 1;
+    var userWinRate = ((userWinCount / winLoseRoundsPlayed) * 100).toFixed(0);
+    var pcWinRate = ((pcWinCount / winLoseRoundsPlayed) * 100).toFixed(0);
+
+    message = `${username} chose ${userSPS} and PC chose ${pcSPS}. ${winOrLose}<br>
+    So far ${username}, you won ${userWinCount}/${winLoseRoundsPlayed} turns and PC won ${pcWinCount}/${winLoseRoundsPlayed} turns. <br><br>
+    Win rate: <br>
+    ${username}- ${userWinRate}%, PC- ${pcWinRate}%. `;
+
+    if (
+      userSPS != "scissors" &&
+      userSPS != "paper" &&
+      userSPS != "stone" &&
+      userSPS != "reversed scissors" &&
+      userSPS != "reversed paper" &&
+      userSPS != "reversed stone"
+    ) {
+      message = `Invalid input. <br> Please select scissors, paper or stone.`;
+    }
```

Comment:
Good input validation. Although this also works, shifting this higher up in the code and doing an early return will make more sense in terms of control flow!